### PR TITLE
Adding paramInput to local input and checks inside optimizer

### LINKED ISSUE #1072: Reading input from paramInput for optimizer
--------
Issue Description
--------

**Is your feature request related to a problem? Please describe.**
GradientBasedOptimizer and its inheritors does not  use the paramInputs, instead it uses the xml node as the input.

**Describe the solution you'd like**
- remove `_readMoreXML` inside Optimizer
- return paramInput as the output of  `_readMoreXMLbase`
- Passing xmlNode and paramInput to `localInputAndChecks`

### PR DESCRIPTION
**Pull request by JiaZhou-PU**
--------
Pull Request Description
--------
##### What issue does this change request address? (Use "#" before the issue to link it, i.e., #42.)
GradientBasedOptimizer has never used the paramInputs, instead, it used the XML as input. Closes #1072 

##### What are the significant changes in functionality due to this change request?
 - Change the default value for the central difference of Finite difference and conjugate gradient into `True` 
 - `Optimizer._readMoreXMLbase` return `paramInput`
 - `localInputAndChecks`, only need `paramInput` as the input, removed `xmlNode` 
 - Add `optIteration` test for the conjugate optimizer
 - Add an elliptic paraboloid analytical model
 - Regold StochasticFiniteDifference test since changing the gradient strategy into the central difference
 - Fix `_identifierToLabel` cdID error,  for divided by 1, modulus will always return 0
----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?
- [x] 9. If any test used as a basis for documentation examples (currently found in `raven/tests/framework/user_guide` and `raven/docs/workshop`) have been changed, the associated documentation must be reviewed and assured the text matches the example.

### CODE REVIEW

**Comment by JiaZhou-PU** [comment]
Finishing param input, needs to add a test for the conjugate gradient. Not ready for review.

**Comment by JiaZhou-PU** [comment]
Waiting on test results on all machine

**Comment by JiaZhou-PU** [comment]
@alfoa Ready for review

**Comment by JiaZhou-PU** [comment]
@alfoa  I removed the readMoreXml inside Optimizer.py in the most recent commit.

File: tests/framework/AnalyticModels/optimizing/paraboloid.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import numpy as np
+
+def run(raven, Inputs):
+  x = raven.x
```
Comment (alfoa) [inline tests/framework/AnalyticModels/optimizing/paraboloid.py]: docstring 

and add a simple (Brief) explaination of the model in here as well (I know you documented it in the analytical test, but also here would be good to have an explaination)

File: tests/framework/Optimizers/optIteration.xml [hunk 2]
```diff
@@ -0,0 +1,80 @@
+<?xml version="1.0" ?>
+<Simulation verbosity="debug">
+  <TestInfo>
+    <name>framework/Optimizers.optIteration</name>
+    <author>ZHOUJ</author>
+    <created>2019-10-09</created>
+    <classesTested>Optimizer</classesTested>
```
Comment (alfoa) [inline tests/framework/Optimizers/optIteration.xml]: which one? the Conjugate

Comment (alfoa) [reply tests/framework/Optimizers/optIteration.xml]: ``` <classesTested>Optimizer.ConjugateGradient</classesTested>```

Comment (alfoa) [reply tests/framework/Optimizers/optIteration.xml]: Is this connected with a requirement? It should be tagged with a requirement tag.

File: tests/framework/AnalyticModels/optimizing/paraboloid.py (hunk 1 above)
Comment (JiaZhou-PU) [reply tests/framework/AnalyticModels/optimizing/paraboloid.py]: Sure

**Comment by JiaZhou-PU** [comment]
Ready for review again. @alfoa

**Comment by alfoa** [comment]
The requirement you should link to this test is ```R-RM-1```

**Comment by alfoa** [comment]
Basically adding the node ```<requirements>R-RM-1</requirements>``` in the <TestInfo>

File: tests/framework/Optimizers/optIteration.xml (hunk 2 above)
Comment (alfoa) [reply tests/framework/Optimizers/optIteration.xml]: ```<requirements>R-RM-1</requirements>```

**Comment by JiaZhou-PU** [comment]
@alfoa  requirement tag added.

**Comment by alfoa** [comment]
Checklist passed...

Comments addressed...

PR can be accepted...

Merging...

**Pull request merged**